### ISSUE TITLE
Allow to omit semicolons for import/export statements

### ISSUE THREAD

**Issue by Voltra** [eslint/eslint#13474]
Is it possible that this could be a configurable option for the 'semi' rule? Eg.

`"omitImportAndExport": true` ignores the semicolons for lines containing only containing imports and exports. There are cases where you want to enforce semicolons in regular code, but may not want/need them for import and export statements for code beautification and since they follow their own 'special syntax' anyways (eg. like classes and function declarations)

_Originally posted by @AaronHarris in https://github.com/eslint/eslint/issues/2174#issuecomment-308911677_

**Comment by anikethsaha** [comment]
Thanks for the issue

I think this is more like a stylistic change and `semi` is a stylistic rule.
No enhancement is being accepted for stylistic rules as per new [policies](https://eslint.org/blog/2020/05/changes-to-rules-policies)

Custom rule or https://github.com/benmosher/eslint-plugin-import can be helpful I guess

**Comment by mdjermanovic** [comment]
Hi @Voltra,

As @anikethsaha mentioned, we are no longer adding new options to stylistic rules.

You can use [eslint-rule-composer](https://github.com/not-an-aardvark/eslint-rule-composer) to make your custom rule that extends the core rule and filters out certain reports.

For example, this rule works like the core `semi` rule but doesn't report any errors for imports:

```js
const ruleComposer = require('eslint-rule-composer');
const eslint = require('eslint');
const semiRule = new eslint.Linter().getRules().get('semi');

module.exports = ruleComposer.filterReports(
  semiRule,
  problem => problem.node.type !== 'ImportDeclaration'
);
```

**Comment by btmills** [comment]
Nice tip on `eslint-rule-composer`! @Voltra if you do decide to create a custom rule for this and don't want to [write a plugin](https://eslint.org/docs/developer-guide/working-with-plugins#rules-in-plugins) for it, you can use the [`--rulesdir` option](https://eslint.org/docs/user-guide/command-line-interface#rulesdir).

I'll close this since we aren't adding stylistic options to core rules anymore, but if you end up customizing it for yourself and need more pointers, reach out [on Discord](https://eslint.org/chat)!
